src/components/team/remove-member-confirm-dialog.tsx

```diff
@@ -0,0 +1,91 @@
+import React, {FunctionComponent} from 'react'
+import {DialogOverlay, DialogContent} from '@reach/dialog'
+import IconX from 'components/icons/icon-x'
+import isObject from 'lodash/isObject'
+import {StateValue} from 'xstate'
+
+type DialogProps = {
+  state: StateValue
```

Comment:
I did try typing this as `state: string | { idle: string }`. It is not simply a string because of the nested states. However even this union type didn't work. XState types this as `StateValue` which I didn't see a way of narrowing to that union type. Declaring it as `StateValue` cleared up the type inference and then I do some explicit runtime checking in `remove-member-confirm-dialog.tsx`.